### PR TITLE
[@types/vuelidate] Add return for async rule

### PR DESCRIPTION
**Pull request by orblazer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://vuelidate.netlify.com/#sub-asynchronous-validation and https://github.com/vuelidate/vuelidate/blob/8ce74dfad262707c4e8b459ee5b8eb5893c742d5/src/index.js#L220
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/vuelidate/vuelidate.d.ts [hunk 1]
```diff
@@ -10,13 +10,16 @@ export const validationMixin: any
 
 // const Validation
 export interface Validation extends Vue {
+    readonly [rule: string]: any // TODO: fix that with boolean
```
Comment (janesser) [inline types/vuelidate/vuelidate.d.ts]: What does it take to resolve this to-do?

**Review by janesser — CHANGES_REQUESTED** [review]
Can you add samples using these features in the test?

File: types/vuelidate/vuelidate.d.ts (hunk 1 above)
Comment (orblazer) [reply types/vuelidate/vuelidate.d.ts]: I am so stupid, this is useless and i have remove that.

**Review by janesser — CHANGES_REQUESTED** [review]
can you please provide some extension to the tests?

That would be great.

**Comment by orblazer** [comment]
@janesser The test for promise return already exist on rule `isUnique`

**Review by ziazon — APPROVED** [review]

**Pull request merged**